Tests/AlgorithmRunner.cs

```diff
@@ -44,6 +44,12 @@ namespace QuantConnect.Tests
     /// </summary>
     public static class AlgorithmRunner
     {
+        /// <summary>
+        /// Dictionary used to store algorithm <see cref="BacktestingResultHandler"/> instances keyed by the algorithm's <see cref="Language"/>
+        /// </summary>
+        public static Dictionary<Language, Dictionary<string, BacktestingResultHandler>> AlgorithmResults = new Dictionary<Language, Dictionary<string, BacktestingResultHandler>>();
```

Comment:
I think this isn't required? Since `BacktestingResultHandlerTests` has his own cache? Don't think we can really trust a global cache for results because for example start and end dates could change between executions

Reply:
We use it to extract the BacktestingResultHandler from the AlgorithmRunner and load it into the BacktestingResultHandlerTests. It will overwrite any existing value (if there is one). The only caching we do is localized within the result handler test fixture.

Reply:
I don't think so. We still need a way to get the BacktestingResultHandler out of the AlgorithmRunner call.

Reply:
Maybe we can create a custom class which holds the `LeanEngineAlgorithmHandlers`, the `AlgorithmManager` and return that instead 🤔